changelog_unreleased/cli/17505.md

```diff
@@ -1,4 +1,4 @@
-#### Fix exitCode when parser can not infer (#17505 by @fisker)
+#### Fix exitCode when parser cannot infer the file type (#17505 by @fisker)
```

Comment:
Other changes looks good, but this one is not.

Reply:
Yes, thank you